Add an explanation on how to use JavaScript chart libraries in MarkBind (#1872)
MarkBind/markbind: docs/userGuide/usingHtmlJavaScriptCss.md

</div>

> ## JavaScript chart libraries

Reviewer: Does this need to be narrowed to chart libraries or write it in general for any JS library but use chart libraries as an example only?

1. Import the library via a CDN or locally.
1. Specify a target HTML element to render the chart.
1. Initialize the chart with the data and options.

<box type="warning">

As mentioned in the [above section](#markdown-in-html), you **should not** leave any blank lines within HTML elements unintentionally, to prevent misinterpreting them as Markdown instead of code/text.
</box>

{{ icon_example }} Here is an example of how to use Chart.js to create a Pie chart.

<include src="codeAndOutput.md" boilerplate>
<variable name="highlightStyle">HTML</variable>
<variable name="code">

<script src="https://cdnjs.cloudflare.com/ajax/libs/Chart.js/3.7.1/chart.min.js"></script>
<div style="width:400px;height:400px;">
  <canvas id="myChart"></canvas>
</div>
<script>
const ctx = document.getElementById('myChart').getContext('2d');
const myChart = new Chart(ctx, {
    type: 'pie',
    data: {
        labels: ['Red', 'Blue', 'Yellow'],
        datasets: [{
            label: '# of Votes',
            data: [12, 19, 3],
            backgroundColor: [
              'rgb(255, 99, 132)',
              'rgb(54, 162, 235)',
              'rgb(255, 205, 86)'
            ]
        }]
    }
});

> </script>

Reviewer 2: I noticed the ECharts example has some helpful comments in the code that correspond to the 3 steps you listed above. Should we add similar comments for this Chart.js example?

</div>

## JavaScript libraries
External JavaScript libraries can be included in MarkBind to add additional & wide-ranging features and functionalities. One such use case is to add a charting library for data visualisation.

> ### Charts

Reviewer 3: Think it would be better with a blank line after the heading to [follow code style!](https://se-education.org/guides/conventions/markdown.html)

Popular chart libraries such as [Chart.js](https://www.chartjs.org/) and [Apache ECharts](https://echarts.apache.org) can be used in MarkBind to create beautiful charts, similar to how they are used in any HTML web page. The details of how to use these libraries are beyond the scope of this section, but you can find more information on their websites. In general, you will perform these 3 steps:
1. Import the library via a CDN or locally.
1. Specify a target HTML element to render the chart.
1. Initialize the chart with the data and options.

<box type="warning">

> As mentioned in the [above section](#markdown-in-html), you **should not** leave any blank lines within HTML elements unintentionally, to prevent misinterpreting them as Markdown instead of code/text.

Reviewer 3: ```suggestion
As mentioned in the [above section](#markdown-in-html), you **should not** leave any blank lines within HTML elements to prevent MarkBind from parsing the contents as Markdown instead of code/text.
```
